contracts/FIFOMarket.sol

```diff
@@ -172,21 +185,21 @@ contract FIFOMarket is
     bytes memory
   ) public override returns (bytes4) {
     for (uint256 i = 0; i < ids.length; i++) {
-      uint256 removalAmount = _removal.balanceOf(address(this), ids[i]);
-      totalActiveSupply += removalAmount;
-      totalNumberActiveRemovals += 1;
-      address supplierAddress = ids[i].supplierAddress();
-      require(
-        _activeSupply[supplierAddress].add(ids[i]),
-        "Market: Removal already in active supply"
-      );
+      uint256 removalToAdd = ids[i];
+      uint256 removalAmount = _removal.balanceOf(address(this), removalToAdd);
+      address supplierAddress = removalToAdd.supplierAddress();
+
+      _activeSupply[supplierAddress].insertByVintage(removalToAdd);
```

Comment:
needs to be asserted since `insertAfter` returns false if it fails

Reply:
e.g., `require(_activeSupply[supplierAddress].insertByVintage(removalToAdd), "...")`

Reply:
Done